eZ/Publish/Core/Limitation/SubtreeLimitationType.php

```diff
@@ -196,6 +196,10 @@ protected function evaluateForContentCreateStruct(APILimitationValue $value, arr
             return false;
         }
 
+        $targets = array_filter($targets, static function ($target): bool {
+            return $target instanceof Location || $target instanceof LocationCreateStruct;
+        });
```

Comment:
Correct me if I am wrong, but it looks like this doesn't do anything relevant. Non-LocationCreateStructs will be ignored anyway because of `if (!$target instanceof LocationCreateStruct) {` few lines below so prefiltering this here changes nothing.

Reply:
@Nattfarinn I don't think that's the case. When we entered `$targets` different than `Location` or `LocationCreateStruct` before we actually had the false flag later on here: https://github.com/ezsystems/ezpublish-kernel/blob/5be6c9a72987af2dbe79561a764c1232f4e3e25a/eZ/Publish/Core/Limitation/SubtreeLimitationType.php#L221
which resulted in `InvalidArgumentException`.

By filtering out such `$targets` this false flag will be encountered only when `$targets` consist of `Location`s, which is a valid step.

Reply:
@barw4 `$hasLocationCreateStruct` is set as default to `false` a line before loop starts, so it will always be `false` unless there's `LocationCreateStruct` in `$targets`. It doesn't matter if you prefilter it or not.

Reply:
@Nattfarinn looks like you are right, that's why I actually passed additional `LocationCreateStructs` here: https://github.com/ezsystems/ezplatform-workflow/pull/111/files#diff-8da4a2807f22c837fdc3c7e7e8b2961cR67, this PR can be closed probably, let me just test this.